x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/accordion_waterfall.tsx

```diff
@@ -0,0 +1,177 @@
+ */
+import { EuiAccordion, EuiAccordionProps } from '@elastic/eui';
+import { Location } from 'history';
+import { isEmpty } from 'lodash';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Margins } from '../../../../../shared/charts/Timeline';
+import { WaterfallItem } from './WaterfallItem';
+import {
+  IWaterfall,
+  IWaterfallItem,
+} from './waterfall_helpers/waterfall_helpers';
+
+interface AccordionWaterfallProps {
+  isOpen: boolean;
+  item: IWaterfallItem;
+  level: number;
+  serviceColors: IWaterfall['serviceColors'];
+  duration: IWaterfall['duration'];
+  waterfallItemId?: string;
+  location: Location;
+  errorsPerTransaction: IWaterfall['errorsPerTransaction'];
+  childrenByParentId: Record<string, IWaterfallItem[]>;
+  onToggleEntryTransaction?: (
+    nextState: EuiAccordionProps['forceState']
+  ) => void;
+  timelineMargins: Margins;
+  onClickWaterfallItem: (item: IWaterfallItem) => void;
+}
+
+const StyledAccordion = styled(EuiAccordion).withConfig({
+  shouldForwardProp: (prop) =>
+    !['childrenCount', 'marginLeftLevel', 'hasError'].includes(prop),
+})<
+  EuiAccordionProps & {
+    childrenCount: number;
+    marginLeftLevel: number;
+    hasError: boolean;
+  }
+>`
+  .euiAccordion {
+    border-top: 1px solid ${({ theme }) => theme.eui.euiColorLightShade};
+  }
+  .euiIEFlexWrapFix {
+    width: 100%;
+    height: 48px;
+  }
+  .euiAccordion__childWrapper {
+    transition: none;
+  }
+
+  .euiAccordion__padding--l {
+    padding-top: 0;
+    padding-bottom: 0;
+  }
+
+  .euiAccordion__iconWrapper {
+    display: flex;
+    position: relative;
+    &:after {
+      content: ${(props) => `'${props.childrenCount}'`};
+      position: absolute;
+      left: 20px;
+      top: -1px;
+      z-index: 1;
+      font-size: ${({ theme }) => theme.eui.euiFontSizeXS};
+    }
+  }
+
+  ${(props) => {
+    const borderLeft = props.hasError
+      ? `2px solid ${props.theme.eui.euiColorDanger};`
+      : `1px solid ${props.theme.eui.euiColorLightShade};`;
+    return `.button_${props.id} {
+      margin-left: ${props.marginLeftLevel}px;
+      border-left: ${borderLeft}
+      &:hover {
+        background-color: ${props.theme.eui.euiColorLightestShade};
+      }
+    }`;
+    //
+  }}
+`;
+
+const WaterfallItemContainer = styled.div`
+  position: absolute;
+  width: 100%;
+  left: 0;
+`;
+
+const getAccordionState = (isOpen: boolean): EuiAccordionProps['forceState'] =>
+  isOpen ? 'open' : 'closed';
+
+export function AccordionWaterfall({
+  isOpen,
+  item,
+  level,
+  serviceColors,
+  duration,
+  childrenByParentId,
+  waterfallItemId,
+  location,
+  errorsPerTransaction,
+  timelineMargins,
+  onClickWaterfallItem,
+}: AccordionWaterfallProps) {
+  const [state, setState] = useState(getAccordionState(isOpen));
+
+  const toggleAccordion = (isAccordionOpen: boolean) => {
+    const newState = getAccordionState(isAccordionOpen);
+    setState(newState);
+  };
```

Comment:
You are still creating state based on a prop. Can we not transform the prop directly without storing it in state? (also, the naming `state` and `setState` is very generic)

Reply:
Or the opposite: have local state that's not based on the prop?